### PR TITLE
List azure resources under resource group in azure_rm_resourcegroup_facts

### LINKED ISSUE #54165: List azure resources under resource group in azure_rm_resourcegroup_facts
##### SUMMARY

Add an option like `show_all_resources` in the `azure_rm_resourcegroup_facts` module, then the module can call API [`ListByResourceGroup()`](https://docs.microsoft.com/en-us/rest/api/resources/resources/listbyresourcegroup) to show all the resources under this group.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

azure_rm_resourcegroup_facts
##### ADDITIONAL INFORMATION

```yaml

```

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

Fixes #54165

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_resourcegroup_facts
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py [hunk 1]
```diff
@@ -55,6 +60,11 @@
...
           - testing
           - foo:bar
+
+    - name: List resources under resource group
+      azure_rm_resourcegroup_facts:
+          name: foo
+          list_resources: yes
 '''
 RETURN = '''
 azure_resourcegroups:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:70]: maybe update return value as well to document new format?

File: lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py [hunk 2]
```diff
@@ -148,6 +164,19 @@ def list_items(self):
...
         return results
 
+    def list_by_rg(self, name):
+        self.log('List resources under resource group')
+        results = []
+        try:
+            response = self.rm_client.resources.list_by_resource_group(name)
+            while True:
+                results.append(response.next().as_dict())
+        except StopIteration:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:209]: why do you need to use StopIteration etc?
I think for item in response will handle the iteration including paged result.

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:209]: The for can only iterate the current page with limited size

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:209]: so if stopiteration that means this page is  done, but there might be other page?

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:209]: Here uses the .next() to query all items in the pages. If use `for`, the code may like
```python
[item.as_dict() for item in response.current_page]
```

**Review by yungezz — APPROVED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py (hunk 2 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py:209]: Well, "for" handles StopIteration exception and Next internally, so you can just have for loop

**Pull request merged**